### ISSUE TITLE
Error loading MAPLIST.BIN

### ISSUE THREAD

**Issue by erasmusa** [planet-x3/px3_ose#2]
When I run the assembled game it throws an error.

![image](https://github.com/user-attachments/assets/2e43823d-7ea6-42e3-818d-76c42aa5720d)

The file is definitely on the FULL1440.IMG disk image and it is possible to view the contents of this file in DosBox. Pressing 'A' to abort continues the game with a the one hard coded map.

Any idea why opening this file would fail?

**Comment by roybaer** [comment]
No, I have not encountered this problem, but maybe we can figure out what is going on.
Can you provide some details on the precise environment?

Such as:

- How did you download the code? Via `git clone` or otherwise?
- Which version did you build? (What does the exit screen say?)
- How did you run DOSBox? Via e.g. `make dosbox_vga` or otherwise?
- Where is the disk image mounted? (With `make dosbox_vga` this should be `A:`)
- What is the current working directory inside DOSBox? (Normally also `A:`)
- What does a complete directory listing look like?

**Comment by M-HT** [comment]
This is a bug in the source code where the filename is missing the zero terminator (in file `diskio.s` - [here](https://github.com/planet-x3/px3_ose/blob/74ccf865812c67083d46671b42ed1f2258f1df3d/src/system/diskio.s#L808)).
The line
`.maplist_name   db "MAPLIST.BIN"`
should be
`.maplist_name   db "MAPLIST.BIN",0`

**Comment by roybaer** [comment]
Good find!
Thank you very much, I will update it shortly.
